frontend/src/components/roboTypeIcon/index.js

```diff
@@ -0,0 +1,29 @@
+</SvgIcon>
+
+const RobotTypeIcon = (props) => {
+  let Icon;
+
+  switch (props.type) {
+    case 'Legged':
```

Comment:
You may want to create an enum for better reusability.

Reply:
done